### PR TITLE
test: "System recreates excess meter when flow is created with the same meterId"

### PR DESCRIPTION
**Pull request by andriidovhan**
covers: #2625

### CODE REVIEW

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy [hunk 1]
```diff
@@ -745,6 +765,57 @@ class FlowCrudSpec extends HealthCheckSpecification {
...
+
+            producer && producer.close()
+        }
+
+        assert northbound.validateSwitch(sw.dpId).meters.excess.size() == amountOfExcessMeters
+
+        when: "Create several flows"
+        List<String> flows = []
+        def amountOfFlows = 3
+        def allowedPorts = topology.getAllowedPortsForSwitch(sw)
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: why is this required? Flow helper should only use allowed ports by default

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy [hunk 2]
```diff
@@ -745,6 +765,57 @@ class FlowCrudSpec extends HealthCheckSpecification {
...
+        amountOfFlows.times {
+            def flow = flowHelper.singleSwitchFlow(sw)
+            flow.source.portNumber = allowedPorts[-1]
+            flow.destination.portNumber = allowedPorts[-2]
+            northbound.addFlow(flow)
+            flows << flow.id
+        }
+
+        then: "Some excess meters should be recreated"
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: I don't quite understand the meaning of this

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy (hunk 1 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: removed

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy (hunk 2 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: renamed
and comment was left

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy [hunk 3]
```diff
@@ -746,6 +766,62 @@ class FlowCrudSpec extends HealthCheckSpecification {
...
+    def "System recreates excess meter when flow is created with the same meterId"() {
+        given: "A Noviflow switch"
+        def sw = topology.activeSwitches.find { it.noviflow } ?: assumeTrue("No suiting switch found", false)
+
+        and: "Create excess meters on the given switch"
+        def fakeBandwidth = 333
+        def amountOfExcessMeters = 10
+        def producer = new KafkaProducer(producerProps)
+        def excessMeterIds = ((MIN_FLOW_METER_ID..100) - northbound.getAllMeters(sw.dpId)
+                .meterEntries*.meterId).take(10)
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: `take(amountOfExcessMeters)`

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy [hunk 4]
```diff
@@ -746,6 +766,62 @@ class FlowCrudSpec extends HealthCheckSpecification {
...
+        northbound.getAllFlows().size() == amountOfFlows
+
+        and: "Needed amount of meter was recreated"
+        // system knows nothing about excess meters
+        // while creating one-switch flow system should create two meters
+        // system checks database and assumes that there is no meter which is used by flow
+        // as a result system creates flow with first available meters (32 and 33)
+        // in reality we've already created excess meters 32..42
+        // excess meters should NOT be just allocated to the created flow
+        // they should be recreated(burst size should be recalculated)
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: for multiline comment:
```
/* I
am
multiline
*/
```

Comment (rtretyak) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: Why we create 10 meters and 3 flows? Why not the same amount?
